### PR TITLE
Worker SDK `14.0.0` upgrade

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description
Upgraded to Worker SDK `14.0.0`:

- Bumped package versions
- Fixed `init.sh/ps1` with all the new package/dll names
- Removed `LocatorFlow` and renamed `AlphaLocatorFlow` to `LocatorFlow`, did a little bit of refactoring as a result.
- Fixed connection flow tests.
- Schema breaking changes in generated code + entity templates.
- Added test case for `ModularUdp` connection type.

#### Tests
Tests are green. Need to test:

- [x] Linux build
- [x] MacOS build
- [x] Android arm64
- [x] Android armv7
- [x] Android x86
- [x] iOS arm
- [x] iOS x86_64

#### Documentation
Changelog 'n' upgrade guide included.

Docs PR --> https://github.com/spatialos/gdk-for-unity/pull/1114

### CODE REVIEW

File: UPGRADE_GUIDE.md [hunk 1]
```diff
@@ -1,5 +1,78 @@
...
+    double y = 2;
+    double z = 3;
+}
+```
+
+And replacing the import of `improbable/vector.schema` and usage of `improbable.Vector3f`/`improbable.Vector3d` with the schema file you defined.
+
+> Note that methods such as `Vector3f.ToUnityVector();` are no longer available and you'll need to reimplement them yourself as extension/static methods. You can find the old implementations here: [`Vector3f`](https://github.com/spatialos/gdk-for-unity/blob/0.2.6/workers/unity/Packages/io.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Partials/Improbable.Vector3f) and [`Vector3d`](https://github.com/spatialos/gdk-for-unity/blob/0.2.6/workers/unity/Packages/io.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Partials/Improbable.Vector3d).
+> 
+> You will be unable to reimplement the operators since C# lacks the ability to define operations via extension methods.
```
Comment (paulbalaji) [inline UPGRADE_GUIDE.md]: could suggest that `Vector3d` is basically the same structure as `Coordinates`?

File: UPGRADE_GUIDE.md [hunk 2]
```diff
@@ -1,5 +1,78 @@
...
+
+#### Connection flow changes
+
+The `AlphaLocatorFlow` and the `LocatorFlow` have been merged. This means that your worker connectors may require some changes. Wherever you were previously using the `AlphaLocatorFlow` or the `ConnectionService.AlphaLocator` enum value, should now be using the `LocatorFlow` and the `ConnectionService.Locator` enum value.
```
Comment (paulbalaji) [inline UPGRADE_GUIDE.md]:
```suggestion
The `AlphaLocatorFlow` and the `LocatorFlow` have been merged. This means that your worker connectors may require some changes. Wherever you were previously using the `AlphaLocatorFlow` or the `ConnectionService.AlphaLocator` enum value, you should now be using the `LocatorFlow` and the `ConnectionService.Locator` enum value.
```

File: workers/unity/Packages/io.improbable.gdk.core/Worker/ConnectionHandlers/ConnectionFlowInitializers.cs [hunk 3]
```diff
@@ -1,12 +1,10 @@
-using System;
 using System.Collections.Generic;
-using Improbable.Worker.CInterop;
 
 namespace Improbable.Gdk.Core
 {
     // TODO: Should this be contravarient?
```
Comment (paulbalaji) [inline workers/unity/Packages/io.improbable.gdk.core/Worker/ConnectionHandlers/ConnectionFlowInitializers.cs]: should we make a ticket for this

File: schema/playground/shared.schema [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package playground;
+
+type Vector3f {
+    float x = 1;
+    float y = 2;
+    float z = 3;
+}
```
Comment (paulbalaji) [inline schema/playground/shared.schema]: newline

File: workers/unity/Packages/io.improbable.gdk.core/Worker/ConnectionHandlers/ConnectionFlowInitializers.cs (hunk 3 above)
Comment (jamiebrynes7) [reply workers/unity/Packages/io.improbable.gdk.core/Worker/ConnectionHandlers/ConnectionFlowInitializers.cs]: I think I'm just gonna remove this TODO

File: UPGRADE_GUIDE.md (hunk 1 above)
Comment (jamiebrynes7) [reply UPGRADE_GUIDE.md]: Added a small sentence

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,12 +5,20 @@
 ### Breaking Changes
 
 - Renamed the public field `AnonymousAuthenticationPort` to `LocatorPort` on the `AlphaLocatorFlow` class and the `RuntimeConfigDefaults` static class. [#1105](https://github.com/spatialos/gdk-for-unity/pull/1105)
+- Upgraded to Worker SDK `14.0.1`. This brings a number of breaking changes [#1112](https://github.com/spatialos/gdk-for-unity/pull/1112)
```
Comment (paulbalaji) [inline CHANGELOG.md]:
```suggestion
- Upgraded to Worker SDK `14.0.1`. This brings a number of breaking changes. [#1112](https://github.com/spatialos/gdk-for-unity/pull/1112)
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,12 +5,20 @@
...
+    - `Vector3f` and `Vector3d` are no longer available in the schema standard library.
+    - The `Improbable.Coordinates.TpSpatialVector3d()` method has been removed.
```
Comment (paulbalaji) [inline CHANGELOG.md]:
```suggestion
    - The `Improbable.Coordinates.ToSpatialVector3d()` method has been removed.
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -5,12 +5,20 @@
...
+    - `LocatorFlow` and `AlphaLocatorFlow` have been merged.
+    - The implementation of the old `LocatorFlow` have been removed.
```
Comment (paulbalaji) [inline CHANGELOG.md]:
```suggestion
    - The implementation of the old `LocatorFlow` has been removed.
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -5,12 +5,20 @@
...
+    - The `ConnectionService.AlphaLocator` enum value has been removed.
+- The following constants have been removed from the `RuntimeConfigNames` static class: `ProjectName`, `SteamDeploymentTag`, and `SteamTicket`.
```
Comment (paulbalaji) [inline CHANGELOG.md]:
```suggestion
- The `ProjectName`, `SteamDeploymentTag`, and `SteamTicket` constants have been removed from the `RuntimeConfigNames` static class.
```

File: UPGRADE_GUIDE.md [hunk 9]
```diff
@@ -1,5 +1,80 @@
 # Upgrade Guide
 
+## From `0.2.6` to `0.2.7`
+
+### Worker SDK `14.0.1` upgrade
+
+There are a number of breaking changes to contend with for the Worker SDK upgrade:
```
Comment (paulbalaji) [inline UPGRADE_GUIDE.md]:
```suggestion
The Worker SDK upgrade introduces breaking changes to the connection flow, and removes the `Vector3f` and `Vector3d` types from the standard schema library.
```

File: UPGRADE_GUIDE.md [hunk 10]
```diff
@@ -1,5 +1,80 @@
...
+
+#### Removal of `Vector3f` and `Vector3d`
+
+These two schema types are no longer available in the schema standard library. You can replace their definitions by defining a schema file in your project:
```
Comment (paulbalaji) [inline UPGRADE_GUIDE.md]:
```suggestion
These two schema types are no longer available in the standard schema library. You can replace their definitions by first defining a schema file in your project:
```

it's what we call it in spos docs 🤷‍♂ 
https://docs.improbable.io/reference/14.0/shared/schema/standard-schema-library

File: UPGRADE_GUIDE.md [hunk 11]
```diff
@@ -1,5 +1,80 @@
...
+}
+
+type Vector3d {
+    double x = 1;
+    double y = 2;
+    double z = 3;
+}
+```
+
+And replacing the import of `improbable/vector.schema` and usage of `improbable.Vector3f`/`improbable.Vector3d` with the schema file you defined.
```
Comment (paulbalaji) [inline UPGRADE_GUIDE.md]:
```suggestion
You should then replace the import of `improbable/vector.schema` and usage of `improbable.Vector3f`/`improbable.Vector3d` with the schema file you defined.
```

**Pull request merged**